test/cql-pytest/test_secondary_index.py

```diff
@@ -100,3 +100,36 @@ def index_used(query, index_name):
         # Local indexes are still preferred over global ones, if they can be used
         index_used(f"SELECT * FROM {table} WHERE p = 1 and v1 = 1 and v3 = 2 and v2 = 2 allow filtering", "my_v2_idx")
         index_used(f"SELECT * FROM {table} WHERE p = 1 and v2 = 1 and v1 = 2 allow filtering", "my_v2_idx")
+
+# Indexes can be created without an explicit name, in which case a default name is chosen.
+# However, due to #8620 it was possible to break the index creation mechanism by creating
+# a properly named regular table, which conflicts with the generated index name.
+# A similar issue still exists for named indexes, so both cases are checked in this scenario.
+def test_create_index_when_its_name_is_taken(scylla_only, cql, test_keyspace):
+    schema = 'p int primary key, v int'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        # In case of default names, a proper name can always eventually be picked
+        index_created = False
+        try:
+            cql.execute(f"CREATE TABLE {table}_v_idx_index (i_do_not_exist_in_the_base_table int primary key)")
+            # Creating an index should succeed, even though its default name is taken
+            # by the table above
+            cql.execute(f"CREATE INDEX ON {table}(v)")
+            index_created = True
+            cql.execute(f"DROP INDEX {table}_v_idx_1")
```

Comment:
Note that you don't really need to drop the index manually - it will be dropped in the end of the test automatically. 
The only reason to have this DROP command is if you want to insist that the automatically chosen table name is table_v_idx_1. But I don't know why we want to insist that. Also this test would fail on Cassnadra (which would use the name table_v_idx, not table_v_idx_1, in this case).